datashuttle/utils/utils.py

```diff
@@ -9,11 +13,30 @@
 # --------------------------------------------------------------------------------------------------------------------
 
 
-def message_user(message: Union[str, list]) -> None:
+def log(message: str) -> None:
+    logger = logging.getLogger("fancylog_")
+    logger.info(message)
+
+
+def log_and_message(message: str, use_rich: bool = False) -> None:
+    log(message)
+    message_user(message, use_rich)
+
+
+def log_and_raise_error(message: str) -> None:
+    logging.error(f"\n\n{' '.join(traceback.format_stack(limit=5))}")
+    logging.error(message)
+    raise_error(message)
+
+
+def message_user(message: Union[str, list], use_rich=False) -> None:
```

Comment:
I don't fully understand why this is necessary, rather than using logging.

Reply:
(discussed above)